packages/sdks/src/functions/register.ts

```diff
@@ -21,6 +22,9 @@ const registry: { [key: string]: any[] } = {};
 export function register(type: 'insertMenu', info: InsertMenuConfig): void;
 export function register(type: string, info: any): void;
 export function register(type: string, info: any) {
+  if (type === 'plugin') {
+    info = serializeIncludingFunctions(info);
+  }
```

Comment:
is there a reason not to always call `serializeIncludingFunctions` for every `register` message?

Reply:
we could do that, but i don't want to break anything existing so felt this was safer. also may be an unnecessary perf overhead if what we have currently works fine

my thoughts were if this becomes whack a mole (e.g. some other register message also needs this) we'd just have it run for everything